helper/intersectionsParsing.js

```diff
@@ -0,0 +1,134 @@
+const _ = require('lodash');
+/*
+this function returns an object that denotes an intersection of form:
+{
+  street1: value1,
+  street2: value2
+}
+*/
+module.exports = function parseIntersections(text) {
+    var str1 = '', str2 = '';
+    if(text.trim().length > 1) {
+        //   var words = text.toLowerCase().split(' ');
+        var words = _.words(text.toLowerCase(), /[^ ]+/g);
+
+        // changes 'e15' to 'East 15', etc.
+        words = directionalSanitizer(words);
+        // changes '6' to '6th', etc
+        words = addOrdinality(words);
+        // only treat input as intersection if contains '&' or 'and'
+        const delimiter = _.includes(text, '&') ? '&' : 'and';
+        const delimiterIndex = words.indexOf(delimiter);
+
+        str1 = wordsToSentence(words, 0, delimiterIndex);
+        str2 = wordsToSentence(words, delimiterIndex+1, words.length);
+    } else {
+        throw 'Missing streets in the intersection';
+    }
+    return { street1: str1, street2: str2 };
+};
+
+// intended for intersections only
+// this function turns '77' into '77th', '3' into '3rd', etc
+function addOrdinality(arr) {
```

Comment:
This function won't handle 11th, 12th, or 13th. There's a pretty elegant implementation at https://github.com/dcousens/ordinal/blob/master/indicator.js. Its up to you if you want to improve this code, use that implementation, or add a requirement on the ordinal package from NPM. Any works for us.

However if you do decide to modify this implementation I would suggest looking at [Array.map()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) which is useful when you are iterating over an array and modifying the values within that array.

Reply:
@orangejulius Oops:) Implemented using the proposed solution. Very elegant.